core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -338,18 +353,32 @@ trait EntityComponent {
       batchInsertEntities(workspaceContext, newEntityRecs.toSeq)
     }
 
-    /** deletes an entity */
-    def delete(workspaceContext: SlickWorkspaceContext, entityType: String, entityName: String): ReadWriteAction[Boolean] = {
+    /** "deletes" entities by hiding and renaming */
+    def hide(workspaceContext: SlickWorkspaceContext, entRefs: Seq[AttributeEntityReference]): ReadWriteAction[Int] = {
       workspaceQuery.updateLastModified(workspaceContext.workspaceId) andThen
-        uniqueResult[EntityRecord](findEntityByName(workspaceContext.workspaceId, entityType, entityName)) flatMap {
-          case None => DBIO.successful(false)
-          case Some(entityRec) =>
-            val deleteActions = deleteEntityAttributes(Seq(entityRec))
-            val deleteEntity = findEntityByName(workspaceContext.workspaceId, entityType, entityName).delete
-            deleteActions andThen deleteEntity.map(_ > 0)
+        lookupEntitiesByNames(workspaceContext.workspaceId, entRefs) flatMap { entRecs =>
+          val hideAction = entRecs map { rec =>
+            hideEntityAttributes(rec.id) andThen
+              hideEntityAction(rec)
+          }
+          DBIO.sequence(hideAction).map(_.sum)
         }
     }
 
+    private def hideEntityAttributes(entityId: Long): ReadWriteAction[Seq[Int]] = {
+      entityAttributes(entityId).result flatMap { attrRecs =>
+        DBIO.sequence(attrRecs map hideEntityAttribute)
+      }
+    }
+
+    private def hideEntityAttribute(attrRec: EntityAttributeRecord): WriteAction[Int] = {
+      entityAttributeQuery.filter(_.id === attrRec.id).map(rec => (rec.deleted, rec.name)).update(true, renameForHiding(attrRec.name))
```

Comment:
The bug was here: I was filtering on attribute name.  So hiding an entity with a "normal" attribute would hide ALL entities' "normal" attributes.  Haha, oops.